tests/lvmo/test_lvm_snapshot_node_reboot.py

```diff
@@ -0,0 +1,307 @@
+import logging
+import time
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import tier2, skipif_lvm_not_installed
+from ocs_ci.framework.testlib import skipif_ocs_version, ManageTest
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.cluster import LVM
+from ocs_ci.ocs.exceptions import (
+    LvSizeWrong,
+    CommandFailed,
+    PodDidNotReachRunningState,
+)
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.ocs.ocp import switch_to_project
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.node import get_node_objs
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_lvm_not_installed
+@skipif_ocs_version("<4.11")
+class TestLvmSnapshotNodeReboot(ManageTest):
+    """
+    Test lvm snapshot with node reboot
+
+    """
+
+    @pytest.mark.parametrize(
+        argnames=["volume_mode", "volume_binding_mode"],
+        argvalues=[
+            pytest.param(
+                *[constants.VOLUME_MODE_FILESYSTEM, constants.WFFC_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[constants.VOLUME_MODE_BLOCK, constants.WFFC_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[
+                    constants.VOLUME_MODE_FILESYSTEM,
+                    constants.IMMEDIATE_VOLUMEBINDINGMODE,
+                ],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+            pytest.param(
+                *[constants.VOLUME_MODE_BLOCK, constants.IMMEDIATE_VOLUMEBINDINGMODE],
+                marks=pytest.mark.polarion_id("OCS-3991"),
+            ),
+        ],
+    )
+    def test_create_snapshot_with_node_reboot(
+        self,
+        nodes,
+        volume_mode,
+        volume_binding_mode,
+        project_factory,
+        lvm_storageclass_factory,
+        snapshot_factory,
+        snapshot_restore_factory,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        test create snapshot reboot
+        .* Create PVC
+        .* Create POD
+        .* Run IO PVC
+        .* Create Snapshot
+        .* Create pvc from Snapshot
+        .* Attach pod
+        .* Reboot node
```

Comment:
please note that the current implementation of node reboot is cold power off. IMO, that is considered a disruption, as it's ungraceful. 
In https://github.com/red-hat-storage/ocs-ci/pull/5382, I wanted to add the option to gracefully restart a node

Reply:
I think it is already implemented for Vsphere:
[https://github.com/red-hat-storage/ocs-ci/blob/55de5d72da977055339e07a4334759786823c121/tests/lvmo/test_lvm_snapshot_node_reboot.py#L218](https://github.com/red-hat-storage/ocs-ci/blob/55de5d72da977055339e07a4334759786823c121/tests/lvmo/test_lvm_snapshot_node_reboot.py#L218)
If you send force=True it will do VM reboot, False will do guest os reboot. I'm doing guest os reboot so I don't think it is disruptive.